app/graphql/types/base_object.rb

```diff
@@ -2,5 +2,30 @@
 
 module Types
   class BaseObject < GraphQL::Schema::Object
+    def self.results_type(&block)
+      wrapped_type = self
+
+      GraphQL::ObjectType.define do
+        type_name = wrapped_type.is_a?(GraphQL::BaseType) ? wrapped_type.name : wrapped_type.graphql_name
+
+        name("#{type_name}Results")
```

Comment:
Cool addition!  I was looking forward to seeing this implementation and I like it.